### PR TITLE
WIP: Aurelia 2 Support

### PR DESCRIPTION
**Pull request by brandonseydel**
Issue:

## What I did
Initial Implementation of Aurelia 2 for storybooks.
## How to test

- Is this testable with Jest or Chromatic screenshots? No.
- Does this need a new example in the kitchen sink apps? Done.
- Does this need an update to the documentation? Included.

If your answer is yes to any of these, please make sure to include it in your PR.

### CODE REVIEW

**Comment by brandonseydel** [comment]
@ndelangen - I am having an issue with knobs working correctly?  Can you see something incorrect?

**Comment by ndelangen** [comment]
@brandonseydel Hey! I'd love to help, shall we schedule a pair programming session?

**Comment by brandonseydel** [comment]
@ndelangen I did solve the issue...missing s on addons.ts doh.  We can still schedule though to tidy things up to the standards you require to be listed as officially supported.

**Comment by ndelangen** [comment]
@brandonseydel yes, I love connecting with contributors, welcome them into the team.

Are you on our discord yet?
https://discord.gg/sMFvFsG

**Review by ndelangen — APPROVED** [review]

**Comment by ndelangen** [comment]
@brandonseydel Is this PR still a WIP?

**Comment by brandonseydel** [comment]
@ndelangen we had it working we just leaving WIP until alpha release.

**Comment by ndelangen** [comment]
@brandonseydel let's get the merge conflicts resolved then?

**Comment by brandonseydel** [comment]
@ndelangen We can do that.  I was trying to keep it up to date, but you guys were making a lot of changes.

**Comment by ndelangen** [comment]
@brandonseydel Yes, we're prepping for a 6.0.0 release.

Optimistically this addition could make it in, IMHO.
It needs an working example added to the /examples folder, and merge conflicts resolved, and then I think it's merge-able.

I trust your expertise in Aurelia.

**Comment by brandonseydel** [comment]
Ok I will have it ready end of week.

**Comment by brandonseydel** [comment]
@ndelangen

File: app/aurelia/src/client/index.ts [hunk 1]
```diff
@@ -0,0 +1,7 @@
+export * from './preview';
+
+// tsc wants to use NodeModule instead of WebpackModule
```
Comment (ndelangen) [inline app/aurelia/src/client/index.ts:3]: Should be fixable by installing @types/webpack-env ?

File: examples/aurelia-kitchen-sink/.storybook/preview.ts [hunk 2]
```diff
@@ -0,0 +1,9 @@
+import { addParameters, addDecorator } from '@storybook/aurelia';
+
+addParameters({
+  options: {
+    brandTitle: 'Aurelia Kitchen Sink',
+    brandUrl: 'https://github.com/storybookjs/storybook/tree/master/examples/aurelia-kitchen-sink',
```
Comment (ndelangen) [inline examples/aurelia-kitchen-sink/.storybook/preview.ts:6]: These should be set in `addons.setConfig` in `manager.js`

File: examples/aurelia-kitchen-sink/src/stories/button.stories.ts [hunk 3]
```diff
@@ -0,0 +1,27 @@
+import { withKnobs, text } from '@storybook/addon-knobs';
+import { action, withActions } from '@storybook/addon-actions';
+import { addComponents, storiesOf } from '@storybook/aurelia';
+import { CoolButton } from '../cool-button/cool-button';
+import 'bootstrap/scss/bootstrap.scss';
+
+storiesOf('Button|Basic', module)
```
Comment (ndelangen) [inline examples/aurelia-kitchen-sink/src/stories/button.stories.ts:7]: Could we use CSF instead here?

Comment (ndelangen) [reply examples/aurelia-kitchen-sink/src/stories/button.stories.ts:7]: The storiesOf format id well tested, already. I think it's safe to assume if CSF works, the storiesOf method will also work.

**Review by ndelangen — APPROVED** [review]
👏 only a few really minor remarks! 👏

File: examples/aurelia-kitchen-sink/src/stories/button.stories.ts (hunk 3 above)
Comment (brandonseydel) [reply examples/aurelia-kitchen-sink/src/stories/button.stories.ts]: Moved into custom elements and changed to CSF

File: examples/aurelia-kitchen-sink/.storybook/preview.ts (hunk 2 above)
Comment (brandonseydel) [reply examples/aurelia-kitchen-sink/.storybook/preview.ts]: Moved

File: app/aurelia/src/client/index.ts (hunk 1 above)
Comment (brandonseydel) [reply app/aurelia/src/client/index.ts]: Added and it didn't complain

**Review by ndelangen — APPROVED** [review]

**Comment by ndelangen** [comment]
There's some linting issues, mind taking care of them?

**Comment by brandonseydel** [comment]
> There's some linting issues, mind taking care of them?
> 
> https://deepscan.io/dashboard/#view=project&tid=4448&pid=6234&bid=51053&subview=pull-request&prid=337259

Done.

**Pull request merged**